pkg/operator/ceph/object/objectstore.go

```diff
@@ -195,6 +195,17 @@ func deleteSingleSiteRealmAndPools(objContext *Context, spec cephv1.ObjectStoreS
 // This is used for quickly getting the name of the realm, zone group, and zone for an object-store to pass into a Context
 func getMultisiteForObjectStore(clusterdContext *clusterd.Context, spec *cephv1.ObjectStoreSpec, namespace, name string) (string, string, string, error) {
 	ctx := context.TODO()
+
+	if spec.IsExternal() {
+		/* *
+		 * Currently external cluster with zones/zonegroup/realm are not supported, will be fixed by https://github.com/rook/rook/issues/6342
+		 * So if user does not create zone/zonegroup, RGW internally creates zone/zonegroup named as `default`, hence Rook can set this value in
+		 * the object context fields.
+		 */
+
+		return "", "default", "default", nil
```

Comment:
How does this affect pre-existing CephObjectStores that were created with zone/zonegroup having the name of the store?

Reply:
@BlaineEXE : The change is applicable only for external ceph clusters with rgw servers running on them, not with the internal cluster in which zone/zonegroup/realm created with name of cephobjectstore. IMO most of the users either create zone/zonegroup/realm with a different name than the name of the ceph object store(since the zone is a kinda collection of cephobject stores make sense to use a different name) or go with none. It is very rare to have users have the same name as cephobjectstore. If the are multiple zone/zonegroup/realm exists the plan is to fix them cleanly by supporting https://github.com/rook/rook/issues/6342.

Reply:
If you say the breaking case is rare, it seems unlikely that we will have any users experience breakage on upgrade to v1.9. In the case of any breakages, we will rely on you to debug those issues at such time.